containers-tests/benchmarks/Map.hs

```diff
@@ -95,6 +98,12 @@ main = do
         , bench "fromDistinctDescList" $ whnf M.fromDistinctDescList elems_rev
         , bench "fromDistinctDescList:fusion" $ whnf (\n -> M.fromDistinctDescList [(i,i) | i <- [n,n-1..1]]) bound
         , bench "minView" $ whnf (\m' -> case M.minViewWithKey m' of {Nothing -> 0; Just ((k,v),m'') -> k+v+M.size m''}) (M.fromAscList $ zip [1..10::Int] [100..110::Int])
+
+        , bench "restrictKeys+withoutKeys"
+        $ whnf (\ks -> M.restrictKeys m ks :*: M.withoutKeys m ks) m_odd_keys
+        , bcompare "/restrictKeys+withoutKeys/"
+        $ bench "partitionKeys"
+        $ whnf (M.partitionKeys m) m_odd_keys
```

Comment:
The `partitionKeys` bench is fine, but keeping around the comparison seems unnecessary.  
It would be good to have two separate benches for `restrictKeys` and `withoutKeys` though, to match `partitionKeys`.

Reply:
Removed comparison and split `restrictKeys+withoutKeys` into two.